pkg/sources/gitlab/gitlab.go

```diff
@@ -241,8 +245,8 @@ func (s *Source) getAllProjects(apiClient *gitlab.Client) ([]*gitlab.Project, er
 	return projectList, nil
 }
 
-func (s *Source) getRepos() ([]*url.URL, []error) {
-	var validRepos []*url.URL
+func (s *Source) getRepos() ([]string, []error) {
+	var validRepos []string
 	var errs []error
 	if len(s.repos) > 0 {
```

Comment:
**Nit (optional):** I prefer structuring code to "fail early" which helps keep nested indentation low.

```go
if len(s.repos) == 0 {
    return nil, nil
}
// rest of logic that was previously inside the `if len(s.repos) > 0` block
```

Reply:
Totally agree.

Reply:
This `if` condition must be inverted.